code/modules/uplink/uplink_devices.dm

```diff
@@ -65,17 +75,45 @@
 	name = "dusty radio"
 	desc = "A dusty looking radio."
 
-/obj/item/uplink/old/Initialize(mapload, owner, tc_amount = 10)
+/obj/item/uplink/old/Initialize(mapload, owner, tc_amount = 10, datum/uplink_handler/uplink_handler_override = null)
 	. = ..()
 	var/datum/component/uplink/hidden_uplink = GetComponent(/datum/component/uplink)
 	hidden_uplink.name = "dusty radio"
 
+// Uplink subtype used as replacement uplink
+/obj/item/uplink/replacement
+	lockable_uplink = TRUE
+
+/obj/item/uplink/replacement/Initialize(mapload, owner, tc_amount = 10, datum/uplink_handler/uplink_handler_override = null)
+	. = ..()
+	var/datum/component/uplink/hidden_uplink = GetComponent(/datum/component/uplink)
+	var/mob/living/replacement_needer = owner
+	if(!istype(replacement_needer))
+		return
+	var/datum/antagonist/traitor/traitor_datum = replacement_needer?.mind.has_antag_datum(/datum/antagonist/traitor)
+	hidden_uplink.unlock_code = traitor_datum?.replacement_uplink_code
+	become_hearing_sensitive()
+
+/obj/item/uplink/replacement/screwdriver_act_secondary(mob/living/user, obj/item/tool)
+	tool.play_tool_sound(src)
+	balloon_alert(user, "deconstructing...")
+	if (!do_after(user, 3 SECONDS, target = src))
+		return FALSE
```

Comment:
```suggestion
	if (!do_after(user, 3 SECONDS, target = src))
		balloon_alert(user, "cancelled!")
		return FALSE
```

although the do_after provides it's own failure UX... feel free to ignore this it just came to my mind

Reply:
I used to consistently do this but Mothblocks told me to stop because the action bar is enough.

Reply:
yeah i had a failure for syncing with the beacon and then mothblocks told me to remove it because the action bar is enough